### PR TITLE
feat(editor): Match parens with CodeMirror

### LINKED ISSUE #816: CodeMirror does not have paren matching
I think this could be a new-contributor friendly CSS fix.

The notebook currently does not provide paren matching — a useful feature that exists in the Jupyter Notebook.

Here's a gif that shows CodeMirror not paren matching in a case where it might be useful.

nteract_no_paren

### PR DESCRIPTION
**Pull request by alainakafkes**
Solution: added automatic paren and quotation matching as well as paren highlighting.

Closes #816.

### CODE REVIEW

**Comment by rgbkrk** [comment]
Oh my gosh, thank you! It makes me laugh every time the easy fixes on CodeMirror are just adding addons. This is so great, thank you!

**Comment by alainakafkes** [comment]
LOL it's alright! Thanks for giving me my first opportunity to contribute to open-source. :) Let me know if anything needs fixing.

**Comment by rgbkrk** [comment]
It looks like there are a few linting touchups to add on which you can find out by running `npm run lint` locally or by viewing [the travis log for our linting check](https://travis-ci.org/nteract/nteract/jobs/164683913).

**Comment by jdetle** [comment]
You're always welcome on [slack](http://slack.nteract.in/), where we keep you apprised of the best coffee making methods, white noise generation, and discuss ideas for where to take nteract!

File: src/notebook/components/cell/editor/index.js [hunk 1]
```diff
@@ -175,8 +183,8 @@ export default class Editor extends React.Component {
             store.getState().app.executionState === 'idle') {
           editor.execCommand('autocomplete', { completeSingle: false });
         }
-      });
-  }
+  });
+}
```
Comment (rgbkrk) [inline src/notebook/components/cell/editor/index.js]: These two lines need to be re-indented.

**Comment by rgbkrk** [comment]
Everything else looks spot on here, this definitely closes #816.

**Comment by alainakafkes** [comment]
Ah I'll fix the indentation shortly! Thanks :)

**Review by rgbkrk — APPROVED** [review]

**Comment by rgbkrk** [comment]
Awesome! I've updated the title to match the changelog standard we've adopted to promote this as a feature for our next batch of release notes. Thank you so much, this is going to help me and others during the day so much!

Welcome to the nteract party! 🎉

**Pull request merged**